pkg/nsx/services/inventory/inventory.go

```diff
@@ -176,6 +185,30 @@ func (s *InventoryService) SyncInventoryObject(bufferedKeys sets.Set[InventoryKe
 	return retryKeys, err
 }
 
+func (s *InventoryService) DeleteResource(externalId string, resourceType InventoryType) error {
+	log.Info("Delete inventory resource", "resource_type", resourceType, "external_id", externalId)
+	switch resourceType {
+	case ContainerProject:
+		inventoryObject := s.ProjectStore.GetByKey(externalId)
+		if inventoryObject == nil {
+			return nil
+		}
+		s.prepareDelete(resourceType, externalId, inventoryObject)
+		return s.DeleteContainerProject(externalId, inventoryObject.(*containerinventory.ContainerProject))
```

Comment:
I just notice we have these unimplemented DeleteXXXX functions. May I learn about what's it for? It looks like the current deletion logic is already complete.

Reply:
For project, it seems no such extra logic.  For pod. it's something more than pod need to deleted. (not sure if it's needed for NCP).

Reply:
I want to rename prepareDelete and DeleteXX, it may raise confusion.

Reply:
https://github.com/vmware-tanzu/nsx-operator/pull/1045